src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1116,7 +1116,12 @@ internal SingleLookupResult CheckViability(Symbol symbol, int arity, LookupOptio
                 ? ((AliasSymbol)symbol).GetAliasTarget(basesBeingResolved)
                 : symbol;
 
-            if (WrongArity(symbol, arity, diagnose, options, out diagInfo))
+            // Check for external symbols marked with 'Microsoft.CodeAnalysis.Embedded' attribute
+            if (!this.Compilation.SourceModule.Equals(unwrappedSymbol.ContainingModule) && unwrappedSymbol.IsHiddenByCodeAnalysisEmbeddedAttribute())
```

Comment:
We have tests that cover one condition. Adding more tests now.
@Vlad I'm also removing this, as now I'll disallow binding to embedded symbols in the same module as well, as user should never be able to use embedded attribute (if the compiler needs to generate one).

---
In reply to: [113549567](https://github.com/dotnet/roslyn/pull/18715#discussion_r113549567) [](ancestors = 113549567,113533730)